### PR TITLE
fix: allow issuer identity to cancel pending JoinToIdentity authorizations

### PR DESCRIPTION
**Pull request by VictorVicente**

### CODE REVIEW

File: src/api/procedures/consumeJoinIdentityAuthorization.ts [hunk 1]
```diff
@@ -59,21 +59,28 @@ export async function prepareConsumeJoinIdentityAuthorization(
...
-  { authRequest }: ConsumeJoinIdentityAuthorizationParams
+  { authRequest, accept }: ConsumeJoinIdentityAuthorizationParams
 ): Promise<boolean> {
-  const { target } = authRequest;
+  const { target, issuer } = authRequest;
   const { context } = this;
 
   let condition;
+  let did: string;
+  const fetchDid = async (): Promise<string> => did || (await context.getCurrentIdentity()).did;
```
Comment (shuffledex) [inline src/api/procedures/consumeJoinIdentityAuthorization.ts:69]: why not use getDid instead?

File: src/api/procedures/consumeJoinIdentityAuthorization.ts [hunk 2]
```diff
@@ -59,21 +59,28 @@ export async function prepareConsumeJoinIdentityAuthorization(
...
     const { address } = context.getCurrentAccount();
     condition = address === target.address;
   } else {
-    const { did } = await context.getCurrentIdentity();
+    did = await fetchDid();
     condition = did === target.did;
   }
 
+  if (!accept) {
+    did = await fetchDid();
```
Comment (shuffledex) [inline src/api/procedures/consumeJoinIdentityAuthorization.ts:80]: define {did} just one time

**Review by monitz87 — APPROVED** [review]

**Pull request merged**